src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,776 @@
+
+			var altState = CoreWindow.GetForCurrentThread().GetKeyState(VirtualKey.Menu);
+			bool isAltPressed = (altState & CoreVirtualKeyStates.Down) == CoreVirtualKeyStates.Down;
+
+			var shiftState = CoreWindow.GetForCurrentThread().GetKeyState(VirtualKey.Shift);
+			bool isShiftPressed = (shiftState & CoreVirtualKeyStates.Down) == CoreVirtualKeyStates.Down;
+
+			bool isDirectionPressed = IsDirectionalKey(key);
+
+			bool canReorderItems = false;
+			var treeView = AncestorTreeView;
+			if (AncestorTreeView != null)
+			{
+				canReorderItems = treeView.CanReorderItems;
+			}
+
+			return canReorderItems && isDirectionPressed && isShiftPressed && isAltPressed && !isControlPressed;
+		}
+
+		private void UpdateTreeViewItemVisualState(TreeNodeSelectionState state)
+		{
+			if (state == TreeNodeSelectionState.Selected)
+			{
+				VisualStateManager.GoToState(this, "TreeViewMultiSelectEnabledSelected", false);
+			}
+			else
+			{
+				VisualStateManager.GoToState(this, "TreeViewMultiSelectEnabledUnselected", false);
+			}
+		}
+
+		private void OnCheckToggle(object sender, RoutedEventArgs args)
+		{
+			var treeView = AncestorTreeView;
+			if (treeView != null)
+			{
+				var listControl = treeView.ListControl;
+				int index = listControl.IndexFromContainer(this);
+				var selectionState = CheckBoxSelectionState((CheckBox)sender);
+				listControl.ListViewModel.ModifySelectByIndex(index, selectionState);
+				UpdateTreeViewItemVisualState(selectionState);
+				RaiseSelectionChangeEvents(selectionState == TreeNodeSelectionState.Selected);
+			}
+		}
+
+		private void RaiseSelectionChangeEvents(bool isSelected)
+		{
+			var peer = FrameworkElementAutomationPeer.FromElement(this);
+			if (peer != null)
+			{
+				var treeViewItemPeer = (TreeViewItemAutomationPeer)peer;
+
+				var selectionEvent = isSelected ?
+					AutomationEvents.SelectionItemPatternOnElementAddedToSelection :
+					AutomationEvents.SelectionItemPatternOnElementRemovedFromSelection;
+				treeViewItemPeer.RaiseAutomationEvent(selectionEvent);
+
+				var isSelectedProperty = SelectionItemPatternIdentifiers.IsSelectedProperty;
+				treeViewItemPeer.RaisePropertyChangedEvent(isSelectedProperty, !isSelected, isSelected);
+			}
+		}
+
+		internal void UpdateSelection(bool isSelected)
+		{
+			var treeView = AncestorTreeView;
+			if (treeView != null)
+			{
+				var node = TreeNode;
+				if (node != null)
+				{
+					treeView?.UpdateSelection(node, isSelected);
+				}
+			}
+		}
+
+		internal void UpdateSelectionVisual(TreeNodeSelectionState state)
+		{
+			var treeView = AncestorTreeView;
+			if (treeView != null)
+			{
+				var listControl = treeView.ListControl;
+				if (listControl.IsMultiselect)
+				{
+					UpdateMultipleSelection(state);
+				}
+				else
+				{
+					var node = TreeNode;
+					if (node != null)
+					{
+						var viewModel = listControl.ListViewModel;
+						var isNodeSelected = viewModel.IsNodeSelected(node);
+						if (isNodeSelected != IsSelected)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless conditional.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)